pkg/cmd/operator/serveprobes/scylladbapistatus.go

```diff
@@ -0,0 +1,179 @@
+	genericclioptions.InClusterReflection
+	ServiceName string
+
+	mux        *http.ServeMux
+	kubeClient kubernetes.Interface
+}
+
+func NewScyllaDBAPIStatusOptions(streams genericclioptions.IOStreams) *ScyllaDBAPIStatusOptions {
+	mux := http.NewServeMux()
+
+	return &ScyllaDBAPIStatusOptions{
+		ProbeServerOptions: *NewProbeServerOptions(streams, naming.ScyllaDBAPIStatusProbePort, mux),
+		ClientConfig:       genericclioptions.NewClientConfig("scylla-operator-api-status-probe"),
+		mux:                mux,
+	}
+}
+
+func (o *ScyllaDBAPIStatusOptions) AddFlags(cmd *cobra.Command) {
+	o.ProbeServerOptions.AddFlags(cmd)
+	o.ClientConfig.AddFlags(cmd)
+	o.InClusterReflection.AddFlags(cmd)
+
+	cmd.Flags().StringVarP(&o.ServiceName, "service-name", "", o.ServiceName, "Name of the service corresponding to the managed node.")
+}
+
+func NewScyllaDBAPIStatusCmd(streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewScyllaDBAPIStatusOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:   "scylladb-api-status",
+		Short: "Serves probes based on ScyllaDB API status.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate(args)
+			if err != nil {
+				return err
+			}
+
+			err = o.Complete(args)
+			if err != nil {
+				return err
+			}
+
+			err = o.Run(streams, cmd)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	o.AddFlags(cmd)
+
+	return cmd
+}
+
+func (o *ScyllaDBAPIStatusOptions) Validate(args []string) error {
+	var errs []error
+
+	errs = append(errs, o.ProbeServerOptions.Validate(args))
+	errs = append(errs, o.ClientConfig.Validate())
+	errs = append(errs, o.InClusterReflection.Validate())
+
+	if len(o.ServiceName) == 0 {
+		errs = append(errs, fmt.Errorf("service-name can't be empty"))
+	} else {
+		serviceNameValidationErrs := apimachineryvalidation.NameIsDNS1035Label(o.ServiceName, false)
+		if len(serviceNameValidationErrs) != 0 {
+			errs = append(errs, fmt.Errorf("invalid service name %q: %v", o.ServiceName, serviceNameValidationErrs))
+		}
+	}
+
+	return apierrors.NewAggregate(errs)
+}
+
+func (o *ScyllaDBAPIStatusOptions) Complete(args []string) error {
+	err := o.ProbeServerOptions.Complete(args)
+	if err != nil {
+		return err
+	}
+
+	err = o.ClientConfig.Complete()
+	if err != nil {
+		return err
+	}
+
+	err = o.InClusterReflection.Complete()
+	if err != nil {
+		return err
+	}
+
+	o.kubeClient, err = kubernetes.NewForConfig(o.ProtoConfig)
+	if err != nil {
+		return fmt.Errorf("can't build kubernetes clientset: %w", err)
+	}
+
+	return nil
+}
+
+func (o *ScyllaDBAPIStatusOptions) Run(originalStreams genericclioptions.IOStreams, cmd *cobra.Command) (returnErr error) {
+	klog.Infof("%s version %s", cmd.Name(), version.Get())
+	cliflag.PrintFlags(cmd.Flags())
+
+	stopCh := signals.StopChannel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go func() {
+		<-stopCh
+		cancel()
+	}()
```

Comment:
shouldn't this be moved to RunE https://github.com/scylladb/scylla-operator/pull/1926/commits/96f2155d10202ff4ef83b4e4a606feb226e0fc2d#diff-eef1c869591b4343abeb9c313296516159ffc51d7fcfe72d7ff23b8970b25358R61? Also eliminating the need for two run functions